src/docs/platform-architecture-reference/platform-automation.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Platform automation
+
+slug: platform-automation
+
+description: Describes the automation we run on to manage the OpenShift platform.
+
+keywords: resource, reclamation, tool, automation, security
+
+page_purpose: Describes the automation we run on to manage the OpenShift platform.
+
+audience: technical lead, developer
+
+author: Matt Spencer, Steven Barre
+
+content_owner: Olena Mitovska
+
+sort_order: 6
+---
+
+# Platform Automation
+
+We use some automation on the platform to help manage resource usage, alert about misconfigured objects and to encourage teams to use images that are secure and up to date. 
+
+## Resource reclamation tool
+
+To date, resource utilization (the use of CPU and RAM) on our platforms has not been efficient. For example, almost 50% of CPU capacity on the Silver cluster is reserved by apps, but not actually used. This unused utilization increases costs and impacts our ability to onboard new teams.  
+ 
+To address this, we’ve developed an automated tool that monitors apps on the platform, looking for inactive or broken deployments. 
+
+### Targets of this tool
+ 
+The tool only monitors `deployments` and `deploymentconfigs` in non-production namespaces (dev, test and tools) and there is no impact to production deployments (prod). This tool does not target databases in `statefulsets`.
+
+The tool looks for two types of deployments: 
+- Deployments that are crashing consistently 
+- Deployments that have not changed in over a year
+
```

Comment:
```suggestion

### How it works

Emails are sent to the Product Owner and Technical Leads registered for the Product in the [Registry](https://registry.developer.gov.bc.ca/). If you want to change who gets the emails, go update the contacts in the Registry.

The first tool checks the `lastUpdateTime` of the `Progressing` block in the `status` field of the Deployment or DeploymentConfig. If it is close to a year old, but not over, a warning email will be sent. If the deployment is managed by ArgoCD then a warning email is sent asking for action to be taken. This is because ArgoCD will just scale it back up after the tool scaled it down. If not managed by ArgoCD and the timestamp is over a year then the deployment is scaled to zero replicas and an email is sent.

You can check the timestamp on your deployment in the YAML under `status`.
 
 ```yaml
   conditions:
  - lastTransitionTime: "2021-11-25T17:04:01Z"
    lastUpdateTime: "2021-11-25T17:04:11Z"
    message: replication controller "backup-postgresql-1" successfully rolled out
    reason: NewReplicationControllerAvailable
    status: "True"
    type: Progressing
```

The second tool looks at the restart count of pods. If a pod has more than 100 restarts it looks for the controlling object of the pod, be it Deployment, DeploymentConfig, or StatefulSet. If the object is managed by ArgoCD then a warning email is sent asking for action to be taken. This is because ArgoCD will just scale it back up after the tool scaled it down. If not managed by ArgoCD, the object is scaled to zero replicas and an email is sent.
```